cmd/vic-machine/delete/delete.go

```diff
@@ -138,6 +139,17 @@ func (d *Uninstall) Run(cli *cli.Context) (err error) {
 		log.Error(err)
 		return errors.New("delete failed")
 	}
+
+	// compare vch version and vic-machine version
+	installerBuild := version.GetBuild()
+	if vchConfig.Version == nil || !installerBuild.Equal(vchConfig.Version) {
+		if !d.Data.Force {
+			log.Errorf("VCH version %q is different with installer version %s. Specify --force to force delete", vchConfig.Version.ShortVersion(), installerBuild.ShortVersion())
+			return errors.New("delete failed")
+		}
+
+		log.Warnf("VCH version %q is different with installer version %s. Force delete will try best to delete everything found.", vchConfig.Version.ShortVersion(), installerBuild.ShortVersion())
```

Comment:
Change warning text to: Force delete will attempt to remove everything related to the installed VCH.